contracts/SimpleCollections.sol

```diff
@@ -14,50 +15,61 @@ contract SimpleCollections is ITokenURIDescriptor, OwnableUpgradeable {
 	}
 
 	ISwapAndStake public swapAndStake;
-	address public gateway;
-	mapping(bytes32 => Image) public images;
-	mapping(uint256 => uint256) public stakedAmountAtMinted;
+	mapping(address => address) public gateway;
+	mapping(address => mapping(bytes32 => Image)) public propertyImages;
+	mapping(address => mapping(uint256 => uint256)) public stakedAmountAtMinted;
 
 	function initialize() external initializer {
 		__Ownable_init();
 	}
 
+	modifier onlyPropertyAuthor(address _property) {
+		address author = IProperty(_property).author();
+		require(author == _msgSender(), "illegal access");
+		_;
+	}
+
 	function image(
 		uint256 id,
 		address,
 		ISTokensManagerStruct.StakingPositions memory _positions,
 		ISTokensManagerStruct.Rewards memory,
 		bytes32 key
 	) external view returns (string memory) {
-		Image memory img = images[key];
-		uint256 stakedAtMinted = stakedAmountAtMinted[id];
+		Image memory img = propertyImages[_positions.property][key];
+		uint256 stakedAtMinted = stakedAmountAtMinted[_positions.property][id];
 		if (stakedAtMinted > _positions.amount) {
 			return "";
 		}
 		return img.src;
 	}
 
-	function setImages(Image[] memory _images, bytes32[] memory _keys)
-		external
-		onlyOwner
-	{
+	function setImages(
+		ISTokensManagerStruct.StakingPositions memory _positions,
```

Comment:
Shouldn't we use just property address instead of `ISTokensManagerStruct.StakingPositions`?

Reply:
You mean passing `address _property` ?
Initially i was thinking the same. But, I thought I would break the consistency.